sw/device/tests/spi_host_smoketest.c

```diff
@@ -18,54 +21,136 @@
 static_assert(__BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__,
               "This test assumes the target platform is little endian.");
 
-#define SFDP_SIGNATURE 0x50444653
-
 OTTF_DEFINE_TEST_CONFIG();
 
-void read_sfdp(dif_spi_host_t *spi_host) {
-  uint8_t buf[256];
-  dif_spi_host_segment_t segments[] = {
-      {
-          .type = kDifSpiHostSegmentTypeOpcode,
-          .opcode = 0x5a,
-      },
-      {
-          .type = kDifSpiHostSegmentTypeAddress,
-          .address =
-              {
-                  .width = kDifSpiHostWidthStandard,
-                  .mode = kDifSpiHostAddrMode3b,
-                  .address = 0x0,
-              },
-      },
-      {
-          .type = kDifSpiHostSegmentTypeDummy,
-          .dummy =
-              {
-                  .width = kDifSpiHostWidthStandard,
-                  .length = 8,
-              },
-      },
-      {
-          .type = kDifSpiHostSegmentTypeRx,
-          .rx =
-              {
-                  .width = kDifSpiHostWidthStandard,
-                  .buf = buf,
-                  .length = sizeof(buf),
-              },
-      },
+// A data pattern to program into the chip:
+// From: http://www.abrahamlincolnonline.org/lincoln/speeches/gettysburg.htm
+static const char kGettysburgPrelude[256] =
+    "Four score and seven years ago our fathers brought forth on this "
+    "continent, a new nation, conceived in Liberty, and dedicated to the "
+    "proposition that all men are created equal.";
+
+// The SFDP structure we'll read out of the chip.
+typedef struct sfdp {
+  union {
+    struct {
+      spi_flash_testutils_sfdp_header_t header;
+      spi_flash_testutils_parameter_header_t param;
+    };
+    uint8_t data[256];
   };
-  CHECK_DIF_OK(
-      dif_spi_host_transaction(spi_host, 0, segments, ARRAYSIZE(segments)));
+  uint32_t *bfpt;
+} sfdp_t;
+
+dif_spi_host_t spi_host;
+sfdp_t sfdp;
+
+status_t test_software_reset(void) {
+  // The software reset sequence is two transactions: RSTEN followed by RST.
+  dif_spi_host_segment_t op = {
+      .type = kDifSpiHostSegmentTypeOpcode,
+      .opcode = kSpiDeviceFlashOpResetEnable,
+  };
+  TRY(dif_spi_host_transaction(&spi_host, /*cs_id=*/0, &op, 1));
+
+  op.opcode = kSpiDeviceFlashOpResetEnable;
+  TRY(dif_spi_host_transaction(&spi_host, /*cs_id=*/0, &op, 1));
+  return OK_STATUS();
+}
+
+status_t test_read_sfdp(void) {
+  spi_flash_testutils_read_sfdp(&spi_host, 0, sfdp.data, sizeof(sfdp.data));
+  LOG_INFO("SFDP signature is 0x%08x", sfdp.header.signature);
+  CHECK(sfdp.header.signature == kSfdpSignature,
```

Comment:
`TRY_CHECK`?

Reply:
The test program needs to abort if the SFDP signature is incorrect -- no other meaningful test can be performed if the flash fails to identify properly.